src/backend/distributed/utils/maintenanced.c

```diff
@@ -630,6 +635,14 @@ MaintenanceDaemonShmemInit(void)
 }
 
 
+/* MaintenanceDaemonSigTermHandler calls proc_exit(0) */
+static void
```

Comment:
Previous callback `die` was leaving worker slots with `in_use` set (see `src/backend/postmaster/bgworker.c` in postgres). If calling `proc_exit` in a signal handler is deemed unclean then we should set a got_SIGTERM to tell main loop to call `proc_exit(0)`